Observer/OrderCancelAfter.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the MIT License
+ * It is available through the world-wide-web at this URL:
+ * https://tldrlegal.com/license/mit-license
+ * If you are unable to obtain it through the world-wide-web, please send an email
+ * to support@buckaroo.nl so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this module to newer
+ * versions in the future. If you wish to customize this module for your
+ * needs please contact support@buckaroo.nl for more information.
+ *
+ * @copyright Copyright (c) Buckaroo B.V.
+ * @license   https://tldrlegal.com/license/mit-license
+ */
+
+namespace Buckaroo\Magento2\Observer;
+
+use Buckaroo\Magento2\Logging\Log;
+use Buckaroo\Magento2\Model\Method\AbstractMethod;
+use Buckaroo\Magento2\Model\ConfigProvider\Account;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Encryption\Encryptor;
+
+class OrderCancelAfter implements \Magento\Framework\Event\ObserverInterface
+{
+    /**
+     * @var \Magento\Framework\App\Config\ScopeConfigInterface
+     */
+    protected $scopeConfig;
+
+    protected $client;
+
+    /** @var Encryptor $encryptor */
+    private $encryptor;
+
+    protected $configProviderAccount;
+
+    protected $logging;
+
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        \Buckaroo\Magento2\Gateway\Http\Client\Json $client,
+        Encryptor $encryptor,
+        Account $configProviderAccount,
+        Log $logging
+    ) {
+        $this->scopeConfig           = $scopeConfig;
+        $this->client                = $client;
+        $this->encryptor             = $encryptor;
+        $this->configProviderAccount = $configProviderAccount;
+        $this->logging               = $logging;
+    }
+
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return void
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        /**
+         * @noinspection PhpUndefinedMethodInspection
+         */
+        /* @var $order \Magento\Sales\Model\Order */
+        $order = $observer->getEvent()->getOrder();
+        /**
+         * @noinspection PhpUndefinedMethodInspection
+         */
+        /**
+         * @var $quote \Magento\Quote\Model\Quote $quote
+         */
+        $quote = $observer->getEvent()->getQuote();
+
+        $payment = $order->getPayment();
+
+        $originalKey = $payment->getAdditionalInformation(AbstractMethod::BUCKAROO_ORIGINAL_TRANSACTION_KEY_KEY);
+
+        $this->logging->addDebug(__METHOD__ . '|1|' . $payment->getMethodInstance()->getCode());
+        $this->logging->addDebug('OrderCancelAfter' . '|1|' . $originalKey);
+
+        $cancel_ppe = $this->scopeConfig->getValue(
```

Comment:
pay per email has its own config provider - that should be used
https://github.com/buckaroo-it/Magento2/blob/develop/Model/ConfigProvider/Method/PayPerEmail.php

Reply:
Fixed